core/classes/Util.php

```diff
@@ -611,4 +558,131 @@ public static function replaceAnchorsWithText($data) {
 		}, $data);
 		return $data;
 	}
+
+	public static function getSetting(DB $db, $setting, $fallback = null) {
+		$value = $db->get('settings', array('name', '=', $setting));
+		if ($value->count()) return $value->first()->value;
+		else return $fallback;
+	}
+
+	public static function getDiscordRoleId(DB $db, $group_id) {
+		$discord_role_id = $db->get('group_sync', array('website_group_id', '=', $group_id));
+		if ($discord_role_id->count()) return $discord_role_id->first()->discord_role_id;
+		else return null;
+	}
+
+	public static function getWebsiteGroup(DB $db, $discord_role_id) {
+		$website_group_id = $db->get('group_sync', array('discord_role_id', '=', $discord_role_id));
+		if ($website_group_id->count()) {
+			$group = $db->get('groups', array('id', '=', $website_group_id->first()->website_group_id));
+			if ($group->count()) {
+				$group_array = array();
+				$group_array['group'] = $group->first();
+				$group_array['primary'] = $db->query('SELECT `primary` FROM nl2_group_sync WHERE discord_role_id = ?', array($discord_role_id))->first()->primary;
+				return $group_array;
+			}
+		}
+		return null;
+	}
+	
+	public static function removeDiscordRole($user_query, $group, Language $language) {
```

Comment:
Maybe these Discord related functions would be best placed in their own Discord class